src/App.tsx

```diff
@@ -1,27 +1,85 @@
-import React from 'react';
 import './App.scss';
+import { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
 import { GoodsList } from './GoodsList';
+import 'bulma';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
+
+export const App: React.FC = () => {
+  const [allGoods, getAllGoods] = useState<Good[]>([]);
+  const [isLoaded, setLoaded] = useState(false);
+  const [isSorted, setSorted] = useState(false);
+  const [isFiltred, setFiltred] = useState(false);
+
+  useEffect(() => {
+    if (isLoaded) {
+      getAll()
+        .then(goods => getAllGoods(goods));
+    }
+
+    if (isSorted) {
+      get5First().then(goods => getAllGoods(goods));
+    }
```

Comment:
i don't think you need this setstates and useEffect
make sorting/filtering methods etc in api/goods and use these methods here